File: c#/classes/inheritance/StandardConversionsWithinClassHierarchy.cs
class Value {
  public double Valuation;
  public string Label;

  public Value(double valuation, string label = "") {
    Valuation = valuation;
    Label = label;
  }

  public void Show() {
    System.Console.WriteLine(Label + ": " + Valuation);
  }
}

class Datum: Value {
  public string Description;

  public Datum(double valuation, string label, string description = "") : base(valuation, label) {
    Description = description;
  }

  public new void Show() {
    System.Console.WriteLine("Value: " + Valuation
      + "\nLabel: " + Label
      + "\nDescription: " + Description
    );
  }
}

class StandardConversionsWithinClassHierarchy {
  public static void DisplayByValue(Value value) {
    value.Show();
  }

  public static void DisplayByReference(ref Value value) {
    value.Show();
  }

  public static void Main(string[] args) {
    Datum temperature = new Datum(24, "Temperature in degree Celsius", "The Central European summer day temperature");

    DisplayByValue(temperature);
    // DisplayByReference(ref temperature);
  }
}

File: java/classes/inheritance/StandardConversionsWithinClassHierarchy.java
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class Value {
  public double value;
  public String label;

  public Value(double value_value, String value_label) {
    value = value_value;
    label = value_label;
  }

  public void show() {
    System.out.println(label + ": " + value);
  };
}

class Datum extends Value {
  public String description;

  public Datum(double value_value, String value_label, String value_description) {
    super(value_value, value_label);
    description = value_description;
  }

  public void show() {
    System.out.println("Value: " + value
      + "\nLabel: " + label
      + "\nDescription: " + description
    );
  };
}

class StandardConversionsWithinClassHierarchy {
  public static void display(Value value) {
    value.show();
  }

  public static void main(String[] args) {
    Datum temperature = new Datum(24, "Temperature in degree Celsius", "The Central European summer day temperature");

    display(temperature);
  }
}
